ui-tests/src/test/java/org/openmrs/reference/page/RegistrationPage.java

```diff
@@ -166,6 +167,7 @@ public void clickOnContactInfo() throws InterruptedException {
     }
 
     public void clickOnPhoneNumber() throws InterruptedException {
+	  waiter.until(ExpectedConditions.visibilityOfElementLocated(PHONE_NUMBER_LABEL));
```

Comment:
Why is it that we originally did not have to wait for the visibility of this element, but the tests passed?

Reply:
Ideally there is some lagging while the gecko driver is trying to pick the element of an id of phoneNumber Label hence throwing visibility Exceptions in this new test that am trying to implement  https://github.com/openmrs/openmrs-contrib-qaframework/pull/54. This being a new test which covers E2E bases workflow requires smooth automated workflows

Reply:
If you are waiting to click, then why not `ExpectedConditions .elementToBeClickable `?

Reply:
Am trying to solve visibility of an element id of phoneNumber Label , so the driver is taking long to see it hence throwing  waiting for visibility of element located by By.id: phoneNumberLabel  and that can be achieved by direct passing of the exact element that is not visible by the driver . Then i guess **ExpectedConditions.elementToBeClickable** handles elements that are taking long to be clicked.

Reply:
If it is the `clickOn(PHONE_NUMBER_LABEL)` that results into that failure, then you want the element to be clickable.

Reply:
i thought  guess the element should first be visible before going to be clickable.However i

Reply:
If it is visible, but not yet clickable, would clickOn succeed?

Reply:
Yes  clickOn will  be able to succeed  after having it visible since we are passing an attribute for it to be to continue to the next functionality

Reply:
Take a second look at the documentation of `ExpectedConditions`. An element can be visible but not clickable. That is why you have `ExpectedConditions.elementToBeClickable` separate from `ExpectedConditions.visibilityOfElementLocated`

Reply:
Sure thanks